server/routes/induction/common/previousWorkExperienceDetailController.ts

```diff
@@ -0,0 +1,99 @@
+import createError from 'http-errors'
+import { NextFunction, Request, RequestHandler, Response } from 'express'
+import type { PreviousWorkExperienceDetailForm } from 'inductionForms'
+import type { InductionDto } from 'inductionDto'
+import InductionController from './inductionController'
+import PreviousWorkExperienceDetailView from './previousWorkExperienceDetailView'
+import TypeOfWorkExperienceValue from '../../../enums/typeOfWorkExperienceValue'
+import logger from '../../../../logger'
+
+/**
+ * Abstract controller class defining functionality common to both the Create and Update Induction journeys.
+ */
+export default abstract class PreviousWorkExperienceDetailController extends InductionController {
+  /**
+   * Returns the Previous Work Experience Detail view; suitable for use by the Create and Update journeys.
+   */
+  getPreviousWorkExperienceDetailView: RequestHandler = async (
+    req: Request,
+    res: Response,
+    next: NextFunction,
+  ): Promise<void> => {
+    const { prisonerSummary, inductionDto } = req.session
+    const { typeOfWorkExperience } = req.params
+
+    let previousWorkExperienceType: TypeOfWorkExperienceValue
+    try {
+      previousWorkExperienceType = this.validateInductionContainsPreviousWorkExperienceOfType({
+        req,
+        inductionDto: inductionDto as InductionDto,
+        invalidTypeOfWorkExperienceMessage: `Attempt to get Previous Work Experience Detail view for invalid work experience type ${typeOfWorkExperience} from ${prisonerSummary.prisonNumber}'s Induction`,
+        typeOfWorkExperienceMissingOnInductionMessage: `Attempt to get Previous Work Experience Detail view for work experience type ${typeOfWorkExperience} that does not exist on ${prisonerSummary.prisonNumber}'s Induction`,
+      })
+    } catch (error) {
+      return next(createError(404, `Previous Work Experience type ${typeOfWorkExperience} not found on Induction`))
+    }
+
+    const previousWorkExperienceDetailsForm =
+      req.session.previousWorkExperienceDetailForm ||
+      toPreviousWorkExperienceDetailForm(inductionDto, previousWorkExperienceType)
+    req.session.previousWorkExperienceDetailForm = undefined
+
+    const view = new PreviousWorkExperienceDetailView(
+      prisonerSummary,
+      this.getBackLinkUrl(req),
+      this.getBackLinkAriaText(req),
+      previousWorkExperienceDetailsForm,
+      previousWorkExperienceType,
+      req.flash('errors'),
+    )
+    return res.render('pages/induction/previousWorkExperience/workExperienceDetail', { ...view.renderArgs })
+  }
+
+  /**
```

Comment:
I've used an options object to pass in the arguments, as I wasnt keen on passing in log/warning message text without some kind of context.
IE as normal function args they are positional args, so its not clear what they mean when reading the caller code
By using an options object its almost as good as kotlin's named args 😁 , so there is some context to what the args are for / what they mean